src/common/updating/updating.cpp

```
@@ -88,70 +88,78 @@ namespace updating
                     }
-                    // Do not break, since https://developer.github.com/v3/repos/releases/#list-releases
-                    // doesn't specify the order in which release object appear
-                    github_version = std::move(*extracted_version);
-                    release_object = std::move(potential_release_object);
                 }
-            }
-            else
-            {
-                const auto body = co_await client.request(Uri{ LATEST_RELEASE_ENDPOINT });
-                release_object = json::JsonValue::Parse(body).GetObjectW();
-                if (auto extracted_version = extract_version_from_release_object(release_object))
+                else
+                {
+                    const auto body = client.request(Uri{ LATEST_RELEASE_ENDPOINT }).get();
+                    release_object = json::JsonValue::Parse(body).GetObjectW();
+                    if (auto extracted_version = extract_version_from_release_object(release_object))
+                    {
+                        github_version = *extracted_version;
+                    }
+                }
+
+                if (github_version <= current_version)
                 {
-                    github_version = *extracted_version;
+                    return version_up_to_date{};
                 }
-            }
 
-            if (github_version <= current_version)
+                auto [installer_download_url, installer_filename] = extract_installer_asset_download_info(release_object);
+                return new_version_download_info{ extract_release_page_url(release_object),
+                                                     std::move(github_version),
+                                                     std::move(installer_download_url),
+                                                     std::move(installer_filename) };
+            }
+            catch (...)
             {
```

PowerToys uses centralized logging via `Logger::info/warn/error` from common/logger. The catch-all block silently swallows exceptions without logging. This violates the repo pattern of comprehensive logging for diagnostics. → Add `Logger::error("Update check failed", ...)` in the catch block to maintain consistency with other error handling in PowerToys.